src/router.js

```diff
@@ -37,6 +38,32 @@ const logoutAndRedirect = async (to, from, next) => {
   return next('/login');
 };
 
+const checkAndSubscribeToLobby = async (to, from, next) => {
+  const gameStore = useGameStore();
+  const authStore = useAuthStore();
+  const { gameId } = to.params;
+  
+  if (!authStore.authenticated) {
+    authStore.redirectGameId = +gameId;
```

Comment:
This doesn't feel like a great way to coerce this value imo as you may end up with `NaN` as a value with nonstandard input.
![image](https://github.com/cuttle-cards/cuttle/assets/414475/f04aa383-96c5-4c5b-b2e2-4a3d3d28dc45)

Reply:
Should we use parseInt() then? I believe Number() will also give you NaN in that situation

Reply:
You might need to use something like this:
```
const gameId = parseInt(game.gameId);
if (!Number.isNaN(gameId) && Number.isFinite(gameId)) {
          // valid
}
```
Otherwise throw the error. Paraphrasing a bit here but use your best judgement. 👍🏼